From 09fe53a17f66cdd98158d9b93d9e7c1f0961e580 Mon Sep 17 00:00:00 2001
From: Huang Chaofan <chaofan.huang@nxp.com>
Date: Wed, 5 Sep 2018 20:03:11 +0800
Subject: [PATCH 4550/5242] MLK-19459 VPU: workaround for vpu encoder to drop
 useless data

commit  41856d9d27b0c6e1fc53d345c714bac209b05613 from
https://source.codeaurora.org/external/imx/linux-imx.git

workaround for vpu encoder to drop useless data

Signed-off-by: Huang Chaofan <chaofan.huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c b/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c
index 14cae38..6b498db 100644
--- a/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c
+++ b/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c
@@ -969,6 +969,9 @@ static void report_stream_done(struct vpu_ctx *ctx,  MEDIAIP_ENC_PIC_INFO *pEncP
 			break;
 	}
 
+	if (ctx->forceStop)
+		return;
+
 	if (!list_empty(&This->drv_q)) {
 		down(&This->drv_q_lock);
 
@@ -982,9 +985,7 @@ static void report_stream_done(struct vpu_ctx *ctx,  MEDIAIP_ENC_PIC_INFO *pEncP
 		// Calculate length - the amount of space remaining in output stream buffer
 		length = p_data_req->vb2_buf->planes[0].length;
 		data_mapped = (void *)vb2_plane_vaddr(p_data_req->vb2_buf, 0);
-		if (wptr == rptr && rptr != start)
-			data_length = end - start;
-		else if (rptr < wptr)
+		if (rptr <= wptr)
 			data_length = wptr - rptr;
 		else if (rptr > wptr)
 			data_length = (end - rptr) + (wptr - start);
-- 
1.7.9.5

